lib/pbench/server/database/models/datasets.py

```diff
@@ -560,6 +560,17 @@ def attach(path=None, controller=None, name=None, state=None) -> "Dataset":
             dataset.advance(state)
         return dataset
 
+    @staticmethod
+    def query(**kwargs) -> "Dataset":
+        """
+        Query dataset object based on a given column name of the run document
+        """
+        try:
+            dataset = Database.db_session.query(Dataset).filter_by(**kwargs).first()
```

Comment:
Is it really worth this one line for fetching a `dataset`?  Is there a higher abstraction we should be implementing?

Reply:
This is no longer a valid comment since this code was rebased on commits which addressed this.